### PR TITLE
Add internal API for wallet generation & posting unsigned txs

### PR DESCRIPTION
**Pull request by ldmberman**

### CODE REVIEW

File: src/apps/app_search.erl [hunk 1]
```diff
@@ -21,11 +21,12 @@
 
 %%@doc Start a search node, linking to a supervisor process
 start_link(Args) ->
-	PID = erlang:apply(app_search, start, Args),
-	{ok, PID}.
+	Pid = erlang:apply(app_search, start, Args),
```
Comment (samcamwilliams) [inline src/apps/app_search.erl]: Pls no.

PID == *P* process *ID* entifier.

File: http_post_unsigned_tx_docs.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+- **Method**
+  POST
+
+#### Example Response
+
+A key which can be used to sign transactions via `POST /unsigned_tx`.
+
+```javascript
+{"wallet_key":"..."}
```
Comment (martin-torhage) [inline http_post_unsigned_tx_docs.md]: `key` in the context of wallet is misleading for me. I thought this was one of the keys in the key pair. My suggestion: `wallet_name` or `wallet_secret`

File: http_post_unsigned_tx_docs.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+- **Method**
+  POST
+
+#### Data Parameter (Post body)
+
+```javascript
+{
+    "last_tx": "",    // base64url encoded ID of last transaction made by address
+    "owner": "",      // base64url encoded modulus of wallet making transaction
```
Comment (martin-torhage) [inline http_post_unsigned_tx_docs.md]: Suggestion: base64url encoded SHA256 hash of the wallet's public key.

File: http_post_unsigned_tx_docs.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    "target": "",     // base64url encoded SHA256 hash of recipients modulus
```
Comment (martin-torhage) [inline http_post_unsigned_tx_docs.md]: Suggestion: base64url encoded SHA256 hash of the recipient's wallet's public key for transactions transferring tokens. Empty if the transaction doesn't transfer tokens.

File: http_post_unsigned_tx_docs.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    "quantity": "",   // string representing the amount of sent AR in winston
```
Comment (martin-torhage) [inline http_post_unsigned_tx_docs.md]: Suggestion: Decimal string of amount of AR in winston to transfer to `target`. Empty if the transaction doesn't transfer any AR.

File: src/ar_http_iface_server.erl [hunk 6]
```diff
@@ -713,6 +706,55 @@ get_tx_filename(Hash) ->
 			{ok, Filename}
 	end.
 
+handle_post_tx(TX) ->
+	% Check whether the TX is already ignored, ignore it if it is not
+	% (and then pass to processing steps).
+	case ar_bridge:is_id_ignored(TX#tx.id) of
+		undefined -> {429, <<"Too Many Requests">>};
```
Comment (martin-torhage) [inline src/ar_http_iface_server.erl]: `ar_bridge:is_id_ignored/1` returns either `true` or `false`.

File: src/ar_http_iface_server.erl [hunk 7]
```diff
@@ -713,6 +706,55 @@ get_tx_filename(Hash) ->
...
+		true -> {208, <<"Transaction already processed.">>};
+		false ->
+			ar_bridge:ignore_id(TX#tx.id),
+			case ar_node:get_current_diff(whereis(http_entrypoint_node)) of
+				unavailable -> {503, [], <<"Transaction verification failed.">>};
```
Comment (martin-torhage) [inline src/ar_http_iface_server.erl]: I don't see how this could happen.

File: http_post_unsigned_tx_docs.md (hunk 2 above)
Comment (ldmberman) [reply http_post_unsigned_tx_docs.md]: ✅

File: http_post_unsigned_tx_docs.md (hunk 3 above)
Comment (ldmberman) [reply http_post_unsigned_tx_docs.md]: ✅

File: http_post_unsigned_tx_docs.md (hunk 4 above)
Comment (ldmberman) [reply http_post_unsigned_tx_docs.md]: ✅

File: http_post_unsigned_tx_docs.md (hunk 5 above)
Comment (ldmberman) [reply http_post_unsigned_tx_docs.md]: ✅

**Pull request merged**